Riot/ViewController/RoomSettingsViewController.m

```diff
@@ -407,6 +407,21 @@ - (void)setSelectedRoomSettingsField:(RoomSettingsViewControllerField)selectedRo
     {
         [self dismissFirstResponder];
         
+        // Check whether user allowed to change room info
+        NSDictionary *eventTypes = @{
+                                     @(RoomSettingsViewControllerFieldName): kMXEventTypeStringRoomName,
+                                     @(RoomSettingsViewControllerFieldTopic): kMXEventTypeStringRoomTopic,
+                                     @(RoomSettingsViewControllerFieldAvatar): kMXEventTypeStringRoomAvatar
+                                     };
+        
+        NSString *eventTypeForSelectedField = eventTypes[@(selectedRoomSettingsField)];
+        
+        MXRoomPowerLevels *powerLevels = [mxRoom.state powerLevels];
+        NSInteger oneSelfPowerLevel = [powerLevels powerLevelOfUserWithUserID:self.mainSession.myUser.userId];
+        
+        if (oneSelfPowerLevel < [powerLevels minimumPowerLevelForSendingEventAsStateEvent:eventTypeForSelectedField])
```

Comment:
You should add a check here on `eventTypeForSelectedField` in order to return if it is nil.